pennylane/gradients/hadamard_gradient.py

```diff
@@ -274,11 +274,6 @@ def _expval_hadamard_grad(tape, argnum, aux_wire):
     coeffs = []
 
     gradient_data = []
-    measurements_probs = [
-        idx
-        for idx, m in enumerate(tape.measurements)
-        if isinstance(m, qml.measurements.ProbabilityMP)
-    ]
```

Comment:
Is there a reason why you moved this to after the loop on the trainable.params? Or just to organize the code better?

Reply:
Yeah, just to move it closer to where it is being used. Reading through the code I was a bit confused why it was defined where it was.

Reply:
Makes sense!